crates/miden-objects/src/account/storage/slot/slot_name.rs

```diff
@@ -34,7 +35,7 @@ use crate::{Felt, FieldElement};
 /// - Each component must only consist of the characters `a` to `z`, `A` to `Z`, `0` to `9` or `_`
 ///   (underscore).
 /// - Each component must not start with an underscore.
-#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
 pub struct SlotName {
     name: Cow<'static, str>,
 }
```

Comment:
Similar to the other comment: this should eventually be `StorageSlotName`, right?

Also, (and this is for another PR), would it make sense to store the slot ID together with the name - e.g., something like:

```rust
pub struct SlotName {
    name: Cow<'static, str>,
    id: StorageSlotId,
}
```
And then the ID could be populated on construction.